### DISCUSSION TITLE
Dependencies via FetchContent instead of submodules

### DISCUSSION THREAD

**Discussion by MikePopoloski** [MikePopoloski/slang#611] [General]
I'm thinking about having the build fetch dependencies via cmake's FetchContent command instead of relying on git submodules, which are a bit finicky and require all users to remember to run an extra step. As part of this I'd like to do away with the SLANG_USE_SYSTEM_LIBS option added by @jrudess and instead rely on the new FIND_PACKAGE_ARGS flag added in cmake 3.24 which lets FindContent first try to find_package() installed on the local system.

I think this is much cleaner and allows users the flexibility to have whichever packages they want installed locally instead of vendored within slang but wanted to see if anyone had concerns about this. In particular cmake 3.24 is very new, but I find it's pretty easy to install newer versions of cmake so hopefully people who need to use system installed packages won't have trouble obtaining it.

**Comment by udif** [comment]
I can only add that the version currently shipping with Ubuntu 22.04 (stable) is 3.22.1, so it's either git submodule or a full source install of cmake. I for sure would prefer the former.

EDIT: I just noted they have binaries for Linux.

**Reply by MikePopoloski** [reply]
Yes, cmake has binaries readily available for all major platforms. It's also available as a package in pypi, so you can `pip install cmake` and run it that way too. Also you only need this if you want the newer feature of overriding FetchContent to use system-installed packages (for things like fmtlib). If you're fine using the ones that slang fetches by default then you can stick with a fairly old version of cmake.
